src/Compilers/CSharp/Portable/Errors/CSharpDiagnosticFormatter.cs

```diff
@@ -1,15 +1,29 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Globalization;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.CSharp;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     public class CSharpDiagnosticFormatter : DiagnosticFormatter
     {
-        internal CSharpDiagnosticFormatter()
+        private static readonly string CompilerAnalyzer = typeof(CSharpCompilerDiagnosticAnalyzer).ToString();
+
+        internal CSharpDiagnosticFormatter(bool displayAnalyzer) :
+            base(displayAnalyzer)
         {
         }
 
-        public new static readonly CSharpDiagnosticFormatter Instance = new CSharpDiagnosticFormatter();
+        protected override string GetAnalyzerIdentity(DiagnosticAnalyzer analyzer)
+        {
+            if (analyzer == null)
+            {
+                return CompilerAnalyzer;
```

Comment:
I presume we get a null value as we never invoke SupportedDiagnostics on compiler analyzer, and hence AnalyzerManager.GetAnalyzer(descriptor) returns null?
Why do we need to display name of compiler analyzer here? We would know that any diagnostic without analyzer name is from compiler. We can avoid the override completely then.